pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -269,7 +275,11 @@ func newAp(path []string, m map[string]interface{}) (*Ap, error) {
 	var name string
 	var argsInterface interface{}
 	for key, value := range m {
-		if isAp(path, value.(map[string]interface{})) {
+		valueMap, ok := value.(map[string]interface{})
+		if !ok {
+			continue
```

Comment:
Can we add a comment explaining why this can be discarded?

Reply:
Sorry, I don't feel like this comment is addressing why we can discard it, it's only addressing why we expect this format. Why is encountering an unexpected format not an error condition?

Reply:
You're correct this should be an error condition. Taking a closer look at the code it appears that it was incorrect. I've made some more updates.